test/e2e/ref.go

```diff
@@ -0,0 +1,46 @@
+package e2e
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	kubeset "k8s.io/client-go/kubernetes"
+)
+
+// JenkinsRef represents a Jenkins instance running on an OpenShift server
+type JenkinsRef struct {
+	host string
+	port string
+	// The namespace in which the Jenkins server is running
+	namespace  string
+	uri_tester *Tester
+	t          *testing.T
+}
+
+// NewRef creates a jenkins reference from an OC client
+func NewRef(t *testing.T, kubeClient *kubeset.Clientset, testNamespace string) *JenkinsRef {
+	svc, err := kubeClient.CoreV1().Services(testNamespace).Get(context.Background(), "jenkins", metav1.GetOptions{})
+	if err != nil {
+		t.Fatalf("%#v", err)
+	}
+	serviceIP := svc.Spec.ClusterIP
+	port := svc.Spec.Ports[0].Port
+
+	j := &JenkinsRef{
+		host:       serviceIP,
```

Comment:
I saw this also in the current e2e test, and I was wondering why and how the service IP is used in the logs. Usually, we see the jenkins Route host there; I remember that I click the link there to go watching in the logs in the jenkins blueocean console in my browser.

Reply:
We do a `curl` with the jenkins SA token to download the job log from the Jenkins instance.  When done inside a pod running in the namespace, you can use the service IP vs. the route.

Of course with the service IP, you avoid having to go over the public internet, and you avoid having to worry about DNS being properly set up *IN* the cluster to resolve that address.
The curl runs in a pod that runs in the same namespace as jenkins.

But yes, when you click the link from your browser, you laptop's DNS comes into play and things resolve.